### ISSUE TITLE
python-missing-require does not report true-conditional requirements as missing

### ISSUE THREAD

**Issue by hroncok** [rpm-software-management/rpmlint#1172]
Consider this metadata:

`/usr/lib/python3.12/site-packages/ipython-8.14.0-py3.12.egg-info/requires.txt`

```
backcall
decorator
jedi>=0.16
matplotlib-inline
pickleshare
prompt_toolkit!=3.0.37,<3.1.0,>=3.0.30
pygments>=2.4.0
stack_data
traitlets>=5

[:python_version < "3.10"]
typing_extensions

[:sys_platform != "win32"]
pexpect>4.3

...
```

rpmlint reports:

```
python3-ipython.noarch: W: python-missing-require backcall
python3-ipython.noarch: W: python-missing-require decorator
python3-ipython.noarch: W: python-missing-require jedi
python3-ipython.noarch: W: python-missing-require matplotlib-inline
python3-ipython.noarch: W: python-missing-require pickleshare
python3-ipython.noarch: W: python-missing-require prompt_toolkit
python3-ipython.noarch: W: python-missing-require pygments
python3-ipython.noarch: W: python-missing-require stack_data
python3-ipython.noarch: W: python-missing-require traitlets
```

However, `W: python-missing-require traitlets typing_extensions` is missing.
